### ISSUE TITLE
bug: infinite loop/memory leak

### ISSUE THREAD

**Issue by yevgenypats** [eugeneware/jpeg-js#53]
Hey there, found the following bug/infinite loop/oom that can be reproduced with the following code:

```javascript
const jpeg = require('jpeg-js');
jpeg.decode(Buffer.from('ffd8ffc09dfdb0ffff0e5296bd7fbbc4f9579096bd7fbbfc0e80d50000ffff36fa400100236701bf73ffaf8003a57f097f5e000000008023c4f9579096bd7fbb008000001500b34e8c018fda5212', 'hex'))
```

found using [jsfuzz](https://github.com/fuzzitdev/jsfuzz)

**Comment by patrickhulce** [comment]
Thanks @yevgenypats! :)

**Comment by yevgenypats** [comment]
@patrickhulce sure!:)

**Comment by patrickhulce** [comment]
this has been fixed by #54
